### DISCUSSION TITLE
Source Generated MediatR

### DISCUSSION THREAD

**Discussion by HernanFAR** [LuckyPennySoftware/MediatR#797] [Ideas]
There is a posibility to use a source generator to the implementation of  the IMediator interface? Like the used in this repo https://github.com/martinothamar/Mediator

**Comment by jbogard** [comment]
Yes there is a possibility, I guess use that repo?

**Reply by AlbertoMonteiro** [reply]
I guess he is talking about to include that in this(your) repository.
I guess the only problem to use that is the support for projects, since source generators requires at least net5.
Could you accept a PR for including that in MediatR?

**Reply by jbogard** [reply]
No, sorry, that's not code I want to maintain.

**Reply by Arkensor** [reply]
@jbogard do you intend to update this library to make use of the optimized approach the other library benefits from or do you intend to stay away from using source generators for MediatR in general?

**Reply by jbogard** [reply]
To put it a different way, any code added to MediatR is code *I* have to maintain. I don't want to maintain anything that complex unless I wrote it. I have no need to write that code (none of the use cases affect me), so I won't. 

If you want a source generator MediatR, use that other library. Or wait until I need it, my client needs it, and pays me to write it.

**Reply by KarimovJavoxir** [reply]
This means that this is never gonna happen. Because, I think clients are not worried about microseconds or nanoseconds. Yeah, While Source Generators would improve performance by 40%, I don't think that your conditions will ever match. So, how should be paid so that you consider using source generators instead of dictionary?

**Reply by jbogard** [reply]
I'm self-employed so that would be a contract with my company to build out such functionality.
